django_colortag/models.py

```diff
@@ -60,10 +61,25 @@ def __gt__(self, other):
         else:
             return NotImplemented
 
+    def is_valid_slug(self, slug):
+        """
+        Check if the slug is valid. By default any slug is valid; you might want
+        to override e.g. to implement uniqueness checks
+        """
+        return True
+
     def save(self, *args, **kwargs):
+        assert self.name
+
+        slug_candidate = self.slug or slugify(self.name)
+        for i in range(1000):
+            if self.is_valid_slug(slug_candidate):
+                break
+            else:
```

Comment:
no need to have else here, because above break. At least for me it's cleaner that way.